libc-bottom-half/sources/chdir.c

```diff
@@ -95,9 +95,10 @@ static const char *make_absolute(const char *path) {
     int need_slash = __wasilibc_cwd[cwd_len - 1] == '/' ? 0 : 1;
     size_t alloc_len = cwd_len + path_len + 1 + need_slash;
     if (alloc_len > make_absolute_len) {
-        make_absolute_buf = realloc(make_absolute_buf, alloc_len);
-        if (make_absolute_buf == NULL)
+        char *tmp = realloc(make_absolute_buf, alloc_len);
+        if (tmp == NULL)
             return NULL;
+        make_absolute_buf = tmp;
```

Comment:
Also figured this potential problem when reading the code carefully - in the out of mem case, `make_absolute_buf` would be set to `NULL` but `make_absolute_len` would be still positiive, leading to mem corruption the next the function will be called.